hetmech/bulk_computation.py

```diff
@@ -0,0 +1,58 @@
+import pandas
+import scipy.special
+import scipy.stats
+
+import hetmech.degree_group
+import hetmech.degree_weight
+import hetmech.hetmat
+
+
+def compute_save_dgp(hetmat, metapath, damping=0.5, compression='gzip', delete_intermediates=True):
+    """
+    Compute summary file of combined degree-grouped permutations (DGP). Aggregates across permutations,
+    deleting intermediates if delete_intermediates=True. Saves resulting files as compressed .tsv files
+    using compression method given by compression. Does not recompute previously saved files.
+    """
```

Comment:
Currently the outer loop is metapath and the inner loop is permutation. For efficiency and caching-purposes, we would like the outer loop to be permutation and have running-DGP metrics updated as metapaths get computed.

This can be saved for a future PR.